subprojects/model-core/src/main/java/org/gradle/api/internal/provider/ProviderInternal.java

```diff
@@ -30,6 +31,12 @@
     @Nullable
     Class<T> getType();
 
+    /**
+     * Visits the build dependencies of this provider, if possible.
+     * @return true if the dependencies added (possibly none), false if the build dependencies are unknown.
+     */
+    boolean maybeVisitBuildDependencies(TaskDependencyResolveContext context);
```

Comment:
An enum might have been more useful. Trying to figure out what `true` means in the context of the overridden method is weird.